advanced/Scripts/database_migration/gravity-db.sh

```diff
@@ -39,4 +39,10 @@ upgrade_gravityDB(){
 		sqlite3 "${database}" < "/etc/.pihole/advanced/Scripts/database_migration/gravity/2_to_3.sql"
 		version=3
 	fi
+	if [[ "$version" == "3" ]]; then
+		# This migration script upgrades the gravity and adlist views
+		# implementing necessary changes for per-client blocking
+		sqlite3 "${database}" < "/etc/.pihole/advanced/Scripts/database_migration/gravity/3_to_4.sql"
```

Comment:
Might want to pull `/etc/.pihole/advanced/Scripts/database_migration/gravity` into a constant.